.github/workflows/build.yml

```
@@ -1,7 +1,12 @@
 name: build and deploy docs
 
-on: push
-
+on:
+  schedule:
+    - cron: '0 */2 * * *' # every 2 hours
+  push:
+    branches:
+      - master
```

```suggestion
  pull_request:
```

Sorry for going back and forth on this!

Now that the script is included in this PR, I've noticed that this check is already in the deploy script.  It's probably useful to generate the documentation as part of CI here.  Then you notice if you break something in a PR to `doc-gen`.